### PR TITLE
`VideoAudioPosts`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `VideoAudioPosts`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

Related to #63409

### CODE REVIEW

File: client/blocks/product-purchase-features-list/test/video-audio-posts.jsx [hunk 1]
```diff
@@ -45,28 +50,22 @@ describe( 'VideoAudioPosts should use proper description', () => {
...
 	} );
 
-	[ PLAN_PREMIUM, PLAN_PREMIUM_2_YEARS ].forEach( ( plan ) => {
-		test( `for premium plan ${ plan }`, () => {
-			const comp = shallow( <VideoAudioPosts { ...props } plan={ plan } /> );
-			expect( comp.find( 'PurchaseDetail' ).props().description ).toContain( '13 GB of media' );
-		} );
+	test.each( [ PLAN_PREMIUM, PLAN_PREMIUM_2_YEARS ] )( `for premium plan %s`, ( plan ) => {
+		render( <VideoAudioPosts { ...props } plan={ plan } /> );
+		expect( screen.queryByTestId( 'purchase-detail' ) ).toHaveTextContent( /13 GB of media/ );
```
Comment (noahtallen) [inline client/blocks/product-purchase-features-list/test/video-audio-posts.jsx:65]: Only thought here is that the "13 GB" portion could change over time, so I wonder if we should let it be more flexible

**Review by noahtallen — APPROVED** [review]

File: client/blocks/product-purchase-features-list/test/video-audio-posts.jsx (hunk 1 above)
Comment (tyxla) [reply client/blocks/product-purchase-features-list/test/video-audio-posts.jsx:65]: I think that's fine, considering that it tests what the user sees. If we change it, the test will fail, and we'll need to update it to confirm that the change was intentional.

File: client/blocks/product-purchase-features-list/test/video-audio-posts.jsx [hunk 2]
```diff
@@ -31,8 +34,10 @@ describe( 'VideoAudioPosts basic tests', () => {
 	};
 
 	test( 'should not blow up and have proper CSS class', () => {
-		const comp = shallow( <VideoAudioPosts { ...props } /> );
-		expect( comp.find( 'PurchaseDetail' ).length ).toBe( 1 );
+		const { container } = render( <VideoAudioPosts { ...props } /> );
+		expect( container.firstChild ).toBeVisible();
+		expect( container.firstChild ).toHaveClass( 'product-purchase-features-list__item' );
+		expect( container.firstChild.firstChild ).toBe( screen.queryByTestId( 'purchase-detail' ) );
```
Comment (tyxla) [inline client/blocks/product-purchase-features-list/test/video-audio-posts.jsx:40]: Not sure how valuable this assertion is TBH.

**Review by tyxla — APPROVED** [review]
This looks good to me 🚀

**Pull request merged**